admin/server-timing.php

```diff
@@ -168,3 +183,56 @@ class="large-text code"
 	</p>
 	<?php
 }
+
+/**
+ * Renders a checkbox for enabling output buffering for Server-Timing.
+ *
+ * @since n.e.x.t
+ */
+function perflab_render_server_timing_page_output_buffer_checkbox() {
+	$slug           = 'output_buffering';
+	$field_id       = "server_timing_{$slug}";
+	$field_name     = PERFLAB_SERVER_TIMING_SETTING . '[' . $slug . ']';
+	$description_id = "{$field_id}_description";
+	$has_filter     = has_filter( 'perflab_server_timing_use_output_buffer' );
+	$is_enabled     = perflab_server_timing_use_output_buffer();
+
+	?>
+	<fieldset>
+		<legend class="screen-reader-text">
+			<?php esc_html_e( 'Output Buffering', 'performance-lab' ); ?>
+		</legend>
+		<input
+			type="checkbox"
+			id="<?php echo esc_attr( $field_id ); ?>"
+			name="<?php echo esc_attr( $field_name ); ?>"
+			aria-describedby="<?php echo esc_attr( $description_id ); ?>"
+			<?php disabled( $has_filter ); ?>
+			<?php checked( $is_enabled ); ?>
+		>
+		<label for="<?php echo esc_attr( $field_id ); ?>">
+			<?php esc_html_e( 'Enable output buffering of template rendering.', 'performance-lab' ); ?>
+		</label>
+		<p id="<?php echo esc_attr( $description_id ); ?>" class="description">
+			<?php if ( $has_filter ) : ?>
+				<?php if ( $is_enabled ) : ?>
+					<?php
+					echo wp_kses(
+						__( 'Output buffering has been forcibly enabled via the <code>perflab_server_timing_use_output_buffer</code> filter.', 'performance-lab' ),
+						array( 'code' => array() )
+					);
+					?>
+				<?php else : ?>
+					<?php
+					echo wp_kses(
+						__( 'Output buffering has been forcibly disabled via the <code>perflab_server_timing_use_output_buffer</code> filter.', 'performance-lab' ),
+						array( 'code' => array() )
+					);
+					?>
+				<?php endif; ?>
+			<?php endif; ?>
+			<?php esc_html_e( 'This is needed to capture metrics after headers have been sent and the template is being rendered. Without output buffering, adding a hook that fires during template rendering may result in a PHP notice: "Function Perflab_Server_Timing::register_metric was called incorrectly. The method must be called before or during the perflab_server_timing_send_header action." Note that output buffering may possibly cause an increase in TTFB if the response would be flushed multiple times.', 'performance-lab' ); ?>
```

Comment:
I think this is too detailed and too long for a field description. The error messages that it leads to IMO shouldn't be mentioned here. (Last but not least, if they are, they should be using a placeholder instead of being part of the translation string.)
```suggestion
			<?php esc_html_e( 'This is needed to capture metrics after headers have been sent and the template is being rendered. Note that output buffering may possibly cause an increase in TTFB if the response would be flushed multiple times.', 'performance-lab' ); ?>
```

Another thought: The "after headers have been sent and the template is being rendered" may be easy to misunderstand? Would it be better to say "after headers have been sent and while the template is being rendered"

Reply:
I edited your suggestion to add "while"